src/app/components/todoform/todoform.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from "@angular/core";
+import { FormsModule, FormGroup, FormControl } from "@angular/forms";
```

Comment:
what is the use of { FormsModule, FormGroup, FormControl  these import if you used template driven form.